airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -273,9 +273,10 @@ def _cast_value(declared_field_types: List, field_name: str, field_value):
         target_type = CUSTOM_FIELD_VALUE_TYPE_CAST_REVERSED.get(target_type_name)
 
         if target_type_name == "number":
-            if field_name.endswith("_id"):
-                # do not cast numeric IDs into float, use integer instead
-                target_type = int
+            # do not cast empty strings into float, use 0 instead.
+            field_value = 0 if len(field_value.strip()) == 0 else field_value
```

Comment:
it is different implementation in comparison of what Sherif said:

"It looks like empty strings are being output as doubles. In this case we should output them as nulls and make sure their type is nullable e.g: ['null', 'double'] for all data types:

Reply:
Maybe  is expected type in not a string and value is empty string then we should put it to None

Reply:
['null', 'double'] - it will be mapped to 'null' in our schemas

Reply:
please check if it works such way